pandas/tests/arrays/categorical/test_constructors.py

```diff
@@ -560,6 +560,15 @@ def test_from_codes_neither(self):
         with pytest.raises(ValueError, match=msg):
             Categorical.from_codes([0, 1])
 
+    def test_from_codes_with_nullable_int(self):
+        codes = pd.array([0, 1], dtype="Int64")
+        categories = ["a", "b"]
+
+        result = Categorical.from_codes(codes, categories=categories)
+        expected = Categorical.from_codes(codes.astype(int), categories=categories)
```

Comment:
```suggestion
        expected = Categorical.from_codes(codes.to_numpy(int), categories=categories)
```

(in case in the near future astype will return an extension array)